src/i18n-module.php

```diff
@@ -111,23 +111,27 @@ class Yoast_I18n_v2 {
 	/**
 	 * Class constructor
 	 *
-	 * @param array $args Contains the settings for the class.
+	 * @param array $args                   Contains the settings for the class.
+	 * @param bool $show_translation_box    Whether the translation box should be shown.
 	 */
-	public function __construct( $args ) {
-		if ( ! is_admin() ) {
-			return;
-		}
+	public function __construct( $args, $show_translation_box = true ) {
+		if( $this->is_admin_in_other_language() ) {
+			$this->init( $args );
 
-		$this->locale = $this->get_admin_locale();
-		if ( 'en_US' === $this->locale ) {
-			return;
+			if ( $show_translation_box ) {
+				add_action( $this->hook, array( $this, 'promo' ) );
+			}
 		}
+	}
 
-		$this->init( $args );
-
-		if ( ! $this->hide_promo() ) {
-			add_action( $this->hook, array( $this, 'promo' ) );
-		}
+	/**
+	 * Returns whether admin is loaded and the language is not en_US.
+	 *
+	 * @return bool If admin is loaded and the language is not en_US.
+	 */
+	public function is_admin_in_other_language() {
+		$this->locale = $this->get_admin_locale();
+		return is_admin() && 'en_US' !== $this->locale;
```

Comment:
If you put back the `is_admin` to the constructor this method only contains `en_US === $this->locale`